src/talk/renderer/Viewer/ViewerHandlerText.vue

```diff
@@ -0,0 +1,210 @@
+}
+
+const { content, loading, error } = useFileContent(toRef(() => props.file.filename), 'text')
+</script>
+
+<template>
+	<ViewerHandlerBase :loading="loading" :error="error" :error-description="error">
+		<template #default>
+			<div class="viewer-text">
+				<template v-if="content">
+					<fieldset :aria-label="t('talk_desktop', 'Controls')" class="viewer-text__controls">
+						<fieldset class="viewer-text__layout-switch" :aria-label="t('talk_desktop', 'Layout')">
+							<NcCheckboxRadioSwitch :checked.sync="layout"
+								:aria-label="t('talk_desktop', 'Compact')"
+								value="compact"
+								type="radio"
+								name="layout"
+								button-variant
+								button-variant-grouped="horizontal">
+								<template #icon>
+									<IconFileDocumentOutline :size="20" />
+								</template>
+							</NcCheckboxRadioSwitch>
+							<NcCheckboxRadioSwitch :checked.sync="layout"
+								:aria-label="t('talk_desktop', 'Wide')"
+								value="wide"
+								type="radio"
+								name="layout"
+								button-variant
+								button-variant-grouped="horizontal">
+								<template #icon>
+									<IconFileChartOutline :size="20" style="transform: rotate(90deg) scaleX(-1)" />
+								</template>
+							</NcCheckboxRadioSwitch>
+						</fieldset>
+						<NcButton :aria-label="wrapLabel"
+							:pressed.sync="wrap"
+							:title="wrapLabel"
+							type="tertiary">
+							<template #icon>
+								<IconWrap :size="20" />
+							</template>
+						</NcButton>
+						<NcButton :aria-label="t('talk_desktop', 'Copy content')"
+							:title="t('talk_desktop', 'Copy content')"
+							type="tertiary"
+							@click="copy">
+							<template #icon>
+								<IconCheck v-if="justCopied" :size="20" />
+								<IconContentCopy v-else :size="20" />
+							</template>
+						</NcButton>
+					</fieldset>
+
+					<div :aria-label="t('talk_desktop', 'Read-only text file content')"
+						class="viewer-text__content"
+						:class="[
+							`viewer-text__content--${format}`, {
+								'viewer-text__content--compact': layout === 'compact',
+								'viewer-text__content--wrap': wrap,
+							}]"
+						contenteditable
+						spellcheck="false"
+						@beforeinput.prevent>
+						<code v-if="format === 'code'">{{ content }}</code>
+						<NcRichText v-else :text="content" :use-extended-markdown="format === 'md'" />
+					</div>
+				</template>
+
+				<div v-else class="viewer-text__empty">
+					<NcEmptyContent :name="t('talk_desktop', 'The file is empty')">
+						<template #icon>
+							<IconFileOutline />
+						</template>
+					</NcEmptyContent>
+				</div>
+			</div>
+		</template>
+	</ViewerHandlerBase>
+</template>
+
+<style scoped>
+.viewer-text {
+	background-color: var(--color-main-background);
+	position: relative;
+	overflow: auto;
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	padding: 0 calc(var(--default-grid-baseline) * 2);
+}
+
+.viewer-text__controls {
+	background-color: inherit;
+	width: 900px;
```

Comment:
Maybe better with full width?

Reply:
Small | Large
-- | --
![image](https://github.com/user-attachments/assets/918f745d-997c-42b8-9878-1704728699f8) | ![image](https://github.com/user-attachments/assets/fbc7da32-fbad-4ce4-8dbc-11d162e977ae)

Reply:
Doesn't look good at light theme, with BG color filling the whole area:

<img width="719" alt="image" src="https://github.com/user-attachments/assets/e266e9ce-03ef-4dc0-911b-6af1deb01f68">

Here I applied `max-width: 900px`:
<img width="648" alt="image" src="https://github.com/user-attachments/assets/a8068f47-fe12-44b9-add4-871c88ba8a09">

Reply:
Because when you have full width mode, and you have text background different (e.g: code) the side edges of 900px bar will be seen.

Reply:
8px side paddings seems to be fine.
<img width="704" alt="image" src="https://github.com/user-attachments/assets/81973bd5-3ba7-4f02-9ec7-8c0b6a22873c">

Not sure how it would look like with `format === code`, because it's not working anymore 👀, but IMO substrate layer with BG color should be applied to all kinds of files (respecting compact mode), and then `--dark-background` should go on top of it

Reply:
background was moved from text content only to an entire container to be consistent with the Text app

Reply:
![image](https://github.com/user-attachments/assets/0eeb1a33-1f44-4157-b32e-2a1547c98f41)

Reply:
![image](https://github.com/user-attachments/assets/5ef1336f-20ea-4513-a688-3a9739d1d557)

Reply:
> to be consistent with the Text app

Ah, okay)